### PR TITLE
doc: update docs for centralized logging

### PR DESCRIPTION
**Pull request by aaSharma14**
Update dashboard docs for centralized logging

Signed-off-by: Aashish Sharma <aasharma@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by nizamial09 — APPROVED** [review]
Thank you!

File: doc/mgr/dashboard.rst [hunk 1]
```diff
@@ -1370,6 +1370,37 @@ something like this::
...
+
+    $ ceph config set global mon_cluster_log_to_file true
+
+5. Click on the Daemon Logs tab under Cluster -> Logs.
+
+6. You can find some pre-defined labels there on clicking the Log browser button such as filename,
+   job etc that can help you query the logs at one go.
+
+7. You can query the logs with LogQL for advanced search and perform some
+   calculations as well - https://grafana.com/docs/loki/latest/logql/.
```
Comment (epuertat) [inline doc/mgr/dashboard.rst:1401]: What about moving this to the Cephadm guide, in the monitoring services section: https://docs.ceph.com/en/latest/cephadm/services/monitoring/

File: doc/mgr/dashboard.rst [hunk 2]
```diff
@@ -1370,6 +1370,37 @@ something like this::
     $ ceph config reset 11
 
 
+Centralized Logging in Ceph
+"""""""""""""""""""""""""""
+
+Ceph-Dashboard now provides centralized logging using Loki & Promtail.
+For this, two new services are introduced in the service list (loki & promtail).
+These services are not deployed by default. To enable the centralized logging -
```
Comment (epuertat) [inline doc/mgr/dashboard.rst:1378]: ?
```suggestion
These services are not deployed by default. To enable the centralized logging:
```

**Review by epuertat — CHANGES_REQUESTED** [review]

File: doc/mgr/dashboard.rst [hunk 3]
```diff
@@ -1370,6 +1370,37 @@ something like this::
     $ ceph config reset 11
 
 
+Centralized Logging in Ceph
+"""""""""""""""""""""""""""
+
+Ceph-Dashboard now provides centralized logging using Loki & Promtail.
```
[the remark this answers is not in the document]
Comment (aaSharma14) [reply doc/mgr/dashboard.rst]: Thanks @epuertat , Updated accordingly

File: doc/mgr/dashboard.rst [hunk 4]
```diff
@@ -1370,6 +1370,37 @@ something like this::
...
+For this, two new services are introduced in the service list (loki & promtail).
```
[the remark this answers is not in the document]
Comment (aaSharma14) [reply doc/mgr/dashboard.rst]: Done

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
jenkins test dashboard cephadm

File: doc/cephadm/services/monitoring.rst [hunk 5]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+
+Ceph-Dashboard now provides centralized logging solution using Loki & Promtail. Centralized Log Management (CLM) is a type
+of logging solution system that consolidates all of your log data and pushes it to one central, accessible,
+and easy-to-use interface. Centralized logging is designed to make your life easier.
+Some of the advantages of using centralized logging are:
+
+#. Linear event timeline (instead of 1000s of different logs), easier to analyze issues.
+#. Real-time live log monitoring: with 1000s of logs is impractical to follow logs.
+#. No need to jump to different hosts for troubleshooting, everything in a single location.
+#. Flexible retention policies. With per-daemon logs, log rotation is usually set to a short interval (1-2 weeks) to save disk usage.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:99]:
```suggestion
#. **Flexible retention policies**: with per-daemon logs, log rotation is usually set to a short interval (1-2 weeks) to save disk usage.
```

File: doc/cephadm/services/monitoring.rst [hunk 6]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+
+Centralized Logging in Ceph
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ceph-Dashboard now provides centralized logging solution using Loki & Promtail. Centralized Log Management (CLM) is a type
+of logging solution system that consolidates all of your log data and pushes it to one central, accessible,
+and easy-to-use interface. Centralized logging is designed to make your life easier.
+Some of the advantages of using centralized logging are:
+
+#. Linear event timeline (instead of 1000s of different logs), easier to analyze issues.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:96]:
```suggestion
#. **Linear event timeline**: it easier to troubleshoot issues analyzing a single chain of events than thousands of different logs from a hundred nodes. 
```

File: doc/cephadm/services/monitoring.rst [hunk 7]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+#. Real-time live log monitoring: with 1000s of logs is impractical to follow logs.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:97]:
```suggestion
#. **Real-time live log monitoring**: it is impractical to follow logs from thousands of different sources.
```

File: doc/cephadm/services/monitoring.rst [hunk 8]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+#. No need to jump to different hosts for troubleshooting, everything in a single location.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:98]: Is not this probably covered by the 1st one?

File: doc/cephadm/services/monitoring.rst [hunk 9]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+#. Flexible retention policies. With per-daemon logs, log rotation is usually set to a short interval (1-2 weeks) to save disk usage.
+#. Increased security & backup: logs can contain sensitive information & usage patterns. Additionally, centralized logging allows for HA, etc.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:100]:
```suggestion
#. **Increased security and backup**: logs can contain sensitive information and expose usage patterns. Additionally, centralized logging allows for HA, etc.
```

File: doc/cephadm/services/monitoring.rst [hunk 10]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+
+Centralized Logging in Dashboard is implemented using two new services - ``loki`` & ``promtail``.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:102]: Not a Dashboard only thing.

File: doc/cephadm/services/monitoring.rst [hunk 11]
```diff
@@ -83,6 +83,31 @@ steps below:
...
+
+Loki is like Prometheus, but for logs. Loki is a log aggregation system designed to store and query logs from all your applications and infrastructure
+and can be configured as a Datasource in grafana.
+
+Promtail is an agent which ships the contents of local logs to a private Grafana Loki instance or Grafana Cloud.
```
Comment (epuertat) [inline doc/cephadm/services/monitoring.rst:107]: This is a verbatim quote from Grafana website (but without clear quoting), and ["are quotes protected by copyright?"](https://www.lizerbramlaw.com/2015/10/27/does-copyright-protect-quotes/). So please remove these lines and point to the Grafana website if someone needs clarification about what Loki is.
```suggestion
Centralized Logging in Dashboard is implemented using two new services:
* Loki: briefly describe loki using your own words and you may add a link to the Loki page.
* Promtail: same thing.
```

File: doc/mgr/dashboard.rst [hunk 12]
```diff
@@ -1369,6 +1369,36 @@ something like this::
...
     $ ceph config reset 11
 
+.. _centralized-logging:
+
+Enable Centralized Logging in Dashboard
+"""""""""""""""""""""""""""""""""""""""
+
+To know more about centralized logging in dashboard, see :ref:`cephadm-monitoring-centralized-logs`
+
+1. Create the loki service on any particular host using Create Services option.
```
Comment (epuertat) [inline doc/mgr/dashboard.rst:1379]: Loki is a proper noun so it should be capitalized:
```suggestion
1. Create the Loki service on any particular host using "Create Services" option.
```

File: doc/mgr/dashboard.rst [hunk 13]
```diff
@@ -1369,6 +1369,36 @@ something like this::
...
+
+2. Similarly create the promtail service which will be by default deployed
```
Comment (epuertat) [inline doc/mgr/dashboard.rst:1381]: Similarly Promtail is a proper noun:
```suggestion
2. Similarly create the Promtail service which will be by default deployed 
```

**Review by epuertat — APPROVED** [review]
Once the issues are addressed, I'm good with it. Thanks @aaSharma14 !

File: doc/mgr/dashboard.rst [hunk 14]
```diff
@@ -1369,6 +1369,36 @@ something like this::
     ...
     $ ceph config reset 11
 
+.. _centralized-logging:
+
+Enable Centralized Logging in Dashboard
+"""""""""""""""""""""""""""""""""""""""
+
+To know more about centralized logging in dashboard, see :ref:`cephadm-monitoring-centralized-logs`
```
Comment (anthonyeleven) [inline doc/mgr/dashboard.rst:1377]: s/know/learn/

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

File: doc/cephadm/services/monitoring.rst [hunk 15]
```diff
@@ -83,6 +83,29 @@ steps below:
...
 
+.. _cephadm-monitoring-centralized-logs:
+
+Centralized Logging in Ceph
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ceph now provides centralized logging solution using Loki & Promtail. Centralized Log Management (CLM) is a type
+of logging solution system that consolidates all of your log data and pushes it to one central, accessible,
+and easy-to-use interface. Centralized logging is designed to make your life easier.
+Some of the advantages of using centralized logging are:
```
Comment (anthonyeleven) [inline doc/cephadm/services/monitoring.rst:94]: Ceph now provides centralized logging with Loki & Promtail. Centralized Log Management (CLM) consolidates all  log data and pushes it to a central repository, with an accessible and easy-to-use interface. Centralized logging is designed to make your life easier. Some of the advantages are:

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

File: doc/cephadm/services/monitoring.rst (hunk 15 above)
Comment (aaSharma14) [reply doc/cephadm/services/monitoring.rst]: Thanks @anthonyeleven , Updated the PR

**Comment by aaSharma14** [comment]
jenkins test dashboard cephadm

**Comment by aaSharma14** [comment]
jenkins test dashboard cephadm

**Review by anthonyeleven — APPROVED** [review]

**Pull request merged**